core/client/src/main/java/alluxio/client/netty/NettyUnderFileSystemBlockReader.java

```diff
@@ -80,48 +70,45 @@ public ByteBuffer read(InetSocketAddress address, long ufsFileId, long offset, l
       SingleResponseListener listener = new SingleResponseListener();
       clientHandler.addListener(listener);
 
-      ChannelFuture channelFuture =
-          channel.writeAndFlush(new RPCFileReadRequest(ufsFileId, offset, length)).sync();
-
+      ChannelFuture channelFuture = channel.writeAndFlush(
+          new RPCUnderFileSystemBlockReadRequest(blockId, offset, length, sessionId, noCache));
+      channelFuture = channelFuture.sync();
       if (channelFuture.isDone() && !channelFuture.isSuccess()) {
-        LOG.error("Failed to read ufs file from %s for ufsFilId %d with error %s.",
-            address.toString(), ufsFileId, channelFuture.cause());
+        LOG.error("Failed to write to %s for block %d with error %s.", address.toString(), blockId,
+            channelFuture.cause());
         throw new IOException(channelFuture.cause());
       }
 
       RPCResponse response = listener.get(NettyClient.TIMEOUT_MS, TimeUnit.MILLISECONDS);
 
       switch (response.getType()) {
-        case RPC_FILE_READ_RESPONSE:
```

Comment:
no. that should never happen because our server returns RPC_BLOCK_READ_RESPONSE for RPC_UFS_BLOCK_READ_REQUEST.  All the incorrect response types indicate a bug in the code and is handled in 'default'.